packages/next/src/client/components/errors/attach-hydration-error-state.ts

```diff
@@ -27,7 +28,7 @@ export function attachHydrationErrorState(error: Error) {
   if (reactHydrationDiffSegments) {
     const diff = reactHydrationDiffSegments[1]
     parsedHydrationErrorState = {
-      ...(error as any).details,
+      ...((error as any).details as HydrationErrorState),
```

Comment:
`.details` was `any` so it hid that we assigned `[string]` to `[string, string, string]`